ephios/static/ephios/js/event_copy.js

```diff
@@ -1,27 +1,145 @@
-                    data = JSON.parse(data)
-                    if (Array.isArray(data) && data.length) {
-                        $('#rrule_occurrences').html(data.join("<br>"));
-                        return;
-                    }
+const {createApp, ref, computed} = Vue
+
+function formatDate(date_obj, sep = "-") {
+    let month = date_obj.getMonth() + 1;
+    let day = date_obj.getDate() + 1;
+    return date_obj.getFullYear() + sep + (month < 10 ? "0" : "") + month + sep + (day < 10 ? "0" : "") + day
+}
+
+function formatHourOrZero(time_string, pickHour=false) {
+    if (time_string && pickHour) {
+        return [time_string.slice(0,2), time_string.slice(3,5), 0]
+    }
+    return [0, 0, 0]
+}
+
+document.addEventListener('DOMContentLoaded', (event) => {
+    createApp({
+        setup() {
+            const pickHour = JSON.parse(document.getElementById("pick_hour").value);
+            const DTSTART = ref(formatDate(new Date()))
+            const DTSTART_TIME = ref("00:00")
+            const rules = ref([])
+            const dates = ref([])
+            const weekdays = [gettext("Monday"), gettext("Tuesday"), gettext("Wednesday"), gettext("Thursday"), gettext("Friday"), gettext("Saturday"), gettext("Sunday")]
+            const frequency_strings = [gettext("year"), gettext("month"), gettext("week"), gettext("day")]
+            const frequencies = rrule.Frequency
+            const months = [{id: 1, short: "Jan", long: "January"}, {id: 2, short: "Feb", long: "February"}, {
+                id: 3, short: "Mar", long: "March"
+            }, {id: 4, short: "Apr", long: "April"}, {id: 5, short: "May", long: "May"}, {
+                id: 6, short: "Jun", long: "June"
+            }, {id: 7, short: "Jul", long: "July"}, {id: 8, short: "Aug", long: "August"}, {
+                id: 9, short: "Sep", long: "September"
+            }, {id: 10, short: "Oct", long: "October"}, {id: 11, short: "Nov", long: "November"}, {
+                id: 12, short: "Dec", long: "December"
+            }]
+
+            async function addRule() {
+                rules.value.push({freq: rrule.RRule.WEEKLY, interval: 1, byweekday: []})
+            }
+
+            async function removeRule(rule) {
+                rules.value.splice(rules.value.indexOf(rule), 1)
+            }
+
+            async function addDate() {
+                dates.value.push({date: "", time: ""});
+            }
+
+            async function removeDate(date) {
+                dates.value.splice(dates.value.indexOf(date), 1)
+            }
+
+            async function changeFreq(rule, freq){
+                rule.freq = freq
+                rule.byweekday = []
+                delete rule.bymonthday
+                delete rule.bymonth
+                delete rule.bysetpos
+                delete rule.month_mode
+            }
+
+            function submitForm(event) {
+                event.target.submit();
+            }
+
+            function isRuleValid(rule) {
```

Comment:
this is stricter than the RFC, but I thought it would be suitable for us (e.g. end of recurrence mandatory, days of weeks/month need to be specified instead of being derived from DTSTART)